backend/pkg/api/connect/integration/transform_test.go

```diff
@@ -0,0 +1,293 @@
+	v1alpha1connect "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1/dataplanev1alpha1connect"
+)
+
+var (
+	technicallyATransform = []byte{0x00, 0x61, 0x73, 0x6D}
+	transformTimeout      = 30 * time.Second
+)
+
+type response struct {
+	Transform *adminapi.TransformMetadata `json:"transform"`
+}
+
+func despawnTopic(ctx context.Context, client v1alpha1connect.TopicServiceClient, topic string) error {
+	_, err := client.DeleteTopic(ctx, connect.NewRequest(&v1alpha1.DeleteTopicRequest{
+		Name: topic,
+	}))
+	return err
+}
+
+func spawnTopic(ctx context.Context, partitionCount int32, client v1alpha1connect.TopicServiceClient, topic string) error {
+	_, err := client.CreateTopic(ctx, connect.NewRequest(&v1alpha1.CreateTopicRequest{
+		Topic: &v1alpha1.CreateTopicRequest_Topic{
+			Name:           topic,
+			PartitionCount: &partitionCount,
+			Configs: []*v1alpha1.CreateTopicRequest_Topic_Config{
+				{
+					Name:  "cleanup.policy",
+					Value: kadm.StringPtr("compact"),
+				},
+			},
+		},
+	}))
+	return err
+}
+
+func spawnTransform(cfg config.Server, name, inputTopic string, outputTopics []string, env []adminapi.EnvironmentVariable, wasm []byte) (*adminapi.TransformMetadata, error) {
+	metadataOut := api.TransformMetadata{
+		Name:         name,
+		InputTopic:   inputTopic,
+		OutputTopics: outputTopics,
+		Environment:  env,
+	}
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+
+	metadataPart, err := writer.CreateFormField("metadata")
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.NewEncoder(metadataPart).Encode(metadataOut); err != nil {
+		return nil, err
+	}
+
+	wasmPart, err := writer.CreateFormFile("wasm_binary", "transform.wasm")
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := wasmPart.Write(wasm); err != nil {
+		return nil, err
+	}
+
+	if err := writer.Close(); err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequestWithContext(context.Background(), "PUT", fmt.Sprintf("http://%s:%d/v1alpha1/transforms", cfg.HTTPListenAddress, cfg.HTTPListenPort), body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", writer.FormDataContentType())
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close() // It's important to close the response body when you're done with it
+
+	if resp.StatusCode != http.StatusCreated {
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil { // If there's an error reading the body, report the status code at least
+			return nil, fmt.Errorf("unexpected status code: %d, error reading response body: %v", resp.StatusCode, err)
+		}
+		return nil, fmt.Errorf("unexpected status code: %d, response: %s", resp.StatusCode, string(bodyBytes))
+	}
+
+	var metaResp response
+	if err := json.NewDecoder(resp.Body).Decode(&metaResp); err != nil {
+		return nil, err
+	}
+
+	return metaResp.Transform, nil
+}
+
+func (s *APISuite) TestDeployTransform() {
```

Comment:
Please use the lower dependencies (i.e. kafka client, redpanda admin client, etc) to create the pre-requisites to avoid cascading test failures